--- cssed-pre0.1/scintilla/gtk/makefile.orig	Mon Dec  8 14:21:03 2003
+++ cssed-pre0.1/scintilla/gtk/makefile	Mon Dec 22 22:17:36 2003
@@ -33,9 +33,9 @@
 endif
 
 ifdef DEBUG
-CXXFLAGS=-DDEBUG -g $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS=-DDEBUG $(OPT) $(CXXBASEFLAGS) $(THREADFLAGS)
 else
-CXXFLAGS=-DNDEBUG -Os $(CXXBASEFLAGS) $(THREADFLAGS)
+CXXFLAGS=-DNDEBUG $(OPT) $(CXXBASEFLAGS) $(THREADFLAGS)
 endif
 
 # If explicit setting of GTK1 or GTK2 then use that else look for
